frontend/src/app/featureFlags/FeatureFlagModal.tsx

```diff
@@ -55,23 +71,59 @@ const FeatureFlagModal: React.FC<Props> = ({
 
   return (
     <Modal data-testid="dev-feature-flags-modal" variant="large" isOpen onClose={onClose}>
-      <ModalHeader title="Override Flags" />
+      <ModalHeader title="Feature Flags: Override Flags" />
       <ModalBody>
-        <Content component="h2">Feature Flags</Content>
-        <Content component="p">
-          Feature flags default to the values defined in the dashboard config.
-        </Content>
-        {renderFlags(definedFeatureFlags, dashboardConfig)}
-
-        {devFlags.length > 0 ? (
-          <>
-            <Content component="h2">Dev Flags</Content>
-            <Content component="p">
-              Dev flags default to inactive and can only be changed for the current session.
+        <Tabs activeKey={activeTabKey} onSelect={(_event, tabIndex) => setActiveTabKey(tabIndex)}>
+          <Tab eventKey={0} title={<TabTitleText>Active</TabTitleText>} data-testid="activeFlagTab">
+            <Content style={{ height: '400px', padding: '8px' }}>
+              <Content component="p">
+                Feature flags default to the values defined in the dashboard config from the server.
+              </Content>
+              <Content component="p">
+                Flags that have an indeterminate value are because they are *not* defined in the
+                server.
+              </Content>
+              <h2> Tech Preview Flags </h2>
+              {renderFlags(Object.keys(techPreviewFlags), dashboardConfig)}
+              <h2> Temporary Developer Feature Flags</h2>
+              {renderFlags(Object.keys(devTemporaryFeatureFlags), dashboardConfig)}
             </Content>
-            {renderFlags(devFlags)}
-          </>
-        ) : null}
+          </Tab>
+          {devFlags.length > 0 && (
+            <Tab
+              eventKey={1}
+              title={<TabTitleText>Dev Flags</TabTitleText>}
+              data-testid="devFlagTab"
+            >
+              <Content style={{ height: '400px', padding: '8px' }}>
+                <Content component="p">
+                  Dev flags default to inactive and can only be changed for the current session.
+                </Content>
+                {renderFlags(devFlags)}
+              </Content>
+            </Tab>
+          )}
+          <Tab eventKey={2} title={<TabTitleText>Legacy</TabTitleText>} data-testid="legacyFlagTab">
+            <PageSection hasOverflowScroll>
```

Comment:
We should avoid using `<PageSection>` where it's not a page. Every component has its own use cases and this will break the design guidelines of PatternFly.

Reply:
ok; will remove that and put in a scroll style instead.

Reply:
done